modules/criteoBidAdapter.js

```diff
@@ -33,6 +32,233 @@ const BUNDLE_COOKIE_NAME = 'cto_bundle';
+
+  imp.tagid = bidRequest.adUnitCode;
+  deepSetValue(imp, 'ext', {
+    ...bidRequest.params.ext,
+    ...imp.ext,
+    rwdd: imp.rwdd,
+    floors: getFloors(bidRequest),
+    bidder: {
+      publishersubid: params?.publisherSubId,
+      zoneid: params?.zoneId,
+      uid: params?.uid,
+    },
+  });
+
+  delete imp.rwdd // oRTB 2.6 field moved to ext
+  delete imp.ext.tid // field moved to source
+
+  if (!context.fledgeEnabled && imp.ext.igs?.ae) {
+    delete imp.ext.igs.ae;
+  }
+
+  if (hasVideoMediaType(bidRequest)) {
+    const paramsVideo = bidRequest.params.video;
+    if (paramsVideo !== undefined) {
+      deepSetValue(imp, 'video', {
+        ...imp.video,
+        skip: imp.video.skip || paramsVideo.skip || 0,
+        placement: imp.video.placement || paramsVideo.placement,
+        minduration: imp.video.minduration || paramsVideo.minduration,
+        playbackmethod: imp.video.playbackmethod || paramsVideo.playbackmethod,
+        startdelay: imp.video.startdelay || paramsVideo.startdelay || 0,
+      })
+    }
+    deepSetValue(imp, 'video.ext', {
+      context: bidRequest.mediaTypes.video.context,
+      playersizes: parseSizes(deepAccess(bidRequest, 'mediaTypes.video.playerSize'), parseSize),
+      plcmt: bidRequest.mediaTypes.video.plcmt,
+      poddur: bidRequest.mediaTypes.video.adPodDurationSec,
+      rqddurs: bidRequest.mediaTypes.video.durationRangeSec,
+    })
+  }
+
+  if (imp.native && typeof imp.native.request !== 'undefined') {
+    let requestNative = JSON.parse(imp.native.request);
+
+    // We remove the native asset requirements if we used the bypass to generate the imp
+    const hasAssetRequirements = requestNative.assets &&
+      (requestNative.assets.length !== 1 || Object.keys(requestNative.assets[0]).length);
+    if (!hasAssetRequirements) {
+      delete requestNative.assets;
+    }
+
+    deepSetValue(imp, 'native.request_native', requestNative);
+    delete imp.native.request;
+  }
+
+  return imp;
+}
+
+/**
+ * Builds a request object for the ORTB 2.5 request.
+ *
+ * @param {function} buildRequest - The function for building a request object.
+ * @param {Array} imps - An array of ORTB 2.5 impression objects.
+ * @param {Object} bidderRequest - The bidder request object.
+ * @param {Object} context - The context object.
+ * @returns {Object} The ORTB 2.5 request object.
+ */
+function request(buildRequest, imps, bidderRequest, context) {
+  let request = buildRequest(imps, bidderRequest, context);
+
+  if (typeof request.app !== 'undefined') {
+    // app
+    if (typeof context.publisherId !== 'undefined') {
+      deepSetValue(request, 'app.publisher.id', context.publisherId);
+    }
+  } else {
+    // site
+    if (typeof context.url !== 'undefined') {
+      deepSetValue(request, 'site.page', context.url);
+    }
+    if (typeof context.publisherId !== 'undefined') {
+      deepSetValue(request, 'site.publisher.id', context.publisherId);
+    }
+  }
+
+  if (bidderRequest.eids) {
```

Comment:
this should probably be `request`

Reply:
`eids` are assigned to `bidderRequest` from `bidRequest.userIdAsEids` [here](https://github.com/prebid/Prebid.js/pull/11486/files/f5098b456ff0cd6bf94951d08fb91c61ae4a06db#diff-74bafa18f954c55e3036d5a4f29828357e9f3a536c0a62e7f51b1dedc4971648R401), as it was done before.
Since `bidRequest.userIdAsEids` is not the common eids field, it won't get converted so I'm not sure why it should be `request`?

Reply:
I thought you were intending to translate `request.user.eids` (which is filled in by ortbConverter) into `request.user.ext.eids` - I missed the intermediate step where you set `bidderRequest.eids`. This makes sense if you don't want to use the eids filled in by ortbConverter - if you do, I added a summary of what I think you're duplicating.